beyond-all-reason/Beyond-All-Reason
# Fix/Improve sector_fire

### Work done

The sector_fire function (currently only used by cortrem) was intended to evenly distribute shots fired across its surface area. However, the formula used does not account for the fact that significantly more area needs to be covered near the outer edge of the sector than near the inner edge, particularly with higher max_range_reduction values. As a result, any given point in the inner section of the targeted sector will be significantly more likely to be hit than any given point in the outer section of the targeted sector. These changes to the formula now take this factor into account, leading to a significantly more even distribution of projectiles. Tested on flat ground, uphill, downhill, and with various sets of modifications to tremor weapondefs.




#### Test steps
- [x] Write the steps needed to test out the changes. Include the expected result.
Spawn a tremor, set it to fire somewhere, observe the more even distribution of projectile impacts compared to the live iteration of sector_fire.
